python/eos/plot/plotter_TEST.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import eos
+
+class PlotterObservableVariableTests(unittest.TestCase):
+    "Test handling of argument 'variable' in Plotter.Observable"
+
+    def setUp(self):
+        self.plot_args = {
+                'plot': {
+                    'x': { 'label': r'$q^2$', 'unit': r'$\textnormal{GeV}^2$', 'range': [0.0, 11.63] },
+                    'y': { 'label': r'$d\mathcal{B}/dq^2$',                    'range': [0.0,  5e-3] },
+                    'legend': { 'location': 'lower left' }
+                },
+                'contents': [
+                    {
+                        'label': r'$\ell=e$',
+                        'type': 'observable',
+                        'observable': 'B->Dlnu::dBR/dq2;l=e,q=d',
+                        'variable': 'q2',
+                        'parameters': {
+                            'mass::mu': 1.0,
+                            'mass::tau': 1.0,
+                        },
+                        'color': 'black',
+                        'range': [0.02, 11.63],
+                    },
+                ]
+            }
+
+
+    def test_variabe_key_0(self):
+        "Check valid case: variable is kinemtic"
+
+        eos.plot.Plotter(self.plot_args)
```

Comment:
Does this work on the CI, where there is no X server? Would probably be a good idea to write it to ``/dev/null`` or similar.

Reply:
`make check` gives 

```
plotter.py:1491: UserWarning: Matplotlib is currently using pgf, which is a non-GUI backend, so canno│   48         self.plot_args['contents'][0].pop('variable')
t show the figure.                                                                                                                     │   49         with self.assertRaisesRegex(ValueError,
  plt.show()
```

but passes.